File: utils/twitch_api.js
const axios = require('axios');
const dotenv = require('dotenv');
dotenv.config();

let accessToken = null;
let expiresAt = 0;

async function getTwitchAccessToken() {
    if (accessToken && Date.now() < expiresAt) {
        return accessToken;
    }

    const clientId = process.env.TWITCH_CLIENT_ID;
    const clientSecret = process.env.TWITCH_CLIENT_SECRET;

    const url = "https://id.twitch.tv/oauth2/token";
    const params = new URLSearchParams({
        client_id: clientId,
        client_secret: clientSecret,
        grant_type: "client_credentials"
    });

    try {
        const response = await axios.post(url, params);
        if (response.status === 200) {
            const data = response.data;
            accessToken = data.access_token;
            expiresAt = Date.now() + data.expires_in * 1000;
            return accessToken;
        } else {
            console.error("Erro ao obter token:", response.status, response.statusText);
            return null;
        }
    } catch (error) {
        console.error("Erro ao obter token:", error);
        return null;
    }
}

module.exports = { getTwitchAccessToken };

File: utils/twitch_api.py
import requests
import os
import time

access_token = None
expires_at = 0

def get_twitch_access_token():
    global access_token, expires_at

    if access_token and time.time() < expires_at:
        return access_token

    client_id = os.getenv("TWITCH_CLIENT_ID")
    client_secret = os.getenv("TWITCH_CLIENT_SECRET")

    url = "https://id.twitch.tv/oauth2/token"
    params = {
        "client_id": client_id,
        "client_secret": client_secret,
        "grant_type": "client_credentials"
    }

    response = requests.post(url, params=params)

    if response.status_code == 200:
        data = response.json()
        access_token = data["access_token"]
        expires_at = time.time() + data["expires_in"]
        return access_token
    else:
        print("Erro ao obter token:", response.status_code, response.text)
        return None
